scripts/validate_package_entries/validate_package_entries.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""Validate that package entries in pyproject.toml are up to date.
+
+This script discovers all Python packages in the components/ and pipelines/
+directories and ensures they are properly listed in pyproject.toml under
+tool.setuptools.packages.
+
+Usage:
+    uv run python scripts/validate_package_entries/validate_package_entries.py
+"""
+
+import argparse
+import sys
+import tomllib
+from pathlib import Path
+from typing import Set
```

Comment:
The `Set` import from typing isn't needed in Python 3.9+. You can use builtin `set[str]` directly (like you already do with `list[str]` on lines 90 and 102). Consider updating lines 24, 30, and 66 to use `set[str]` and removing this import for consistency.